power-pay-frontend/src/Hooks/useStorage.ts

```diff
@@ -0,0 +1,47 @@
+// Here we first start by importing the neccassary modules needed to create the Hook.
+
+import { useState, useEffect, useCallback } from "react";
+
+// we define our hook variable specifying its paramaters and types and also the storing logic
+const useStorage = (
+  key: string,
+  defaultValue: number | string | null = null,
+  type?: "local" | "session"
+) => {
+  const storage =
```

Comment:
What if I want to save into the `indexedDb`? For the requirements of this app, we won't use use sessionStorage. For more sensible things, we'll use indexedDb and localStorage for the rest